tests/src/base/test_forcing.py

```diff
@@ -331,6 +334,45 @@ def test_retrieve_caravan_forcing(tmp_path: Path, mock_retrieve: mock.MagicMock)
     mock_retrieve.assert_called_once_with(basin_id.split("_")[0])
 
 
+def test_retrieve_caravan_forcing_empty_vars(
+    tmp_path: Path, mock_retrieve: mock.MagicMock
+):
+    basin_id = "camels_03439000"
+    test_files_dir = Path(__file__).parent / "forcing_files"
+    tmp_camels_dir = tmp_path / "camels"
+    copytree(test_files_dir, tmp_camels_dir)
+    caravan_forcing = CaravanForcing.generate(
+        start_time="1981-01-01T00:00:00Z",
+        end_time="1981-03-01T00:00:00Z",
+        directory=str(tmp_camels_dir),
+        basin_id=basin_id,
+    )
+    caravan_forcing.save()
+    ds = caravan_forcing.to_xarray()
+    content = list(ds.data_vars.keys())
+    expected = ["Q", "evspsblpot", "pr", "tas", "tasmax", "tasmin"]
+    assert content == expected
+    mock_retrieve.assert_called_once_with(basin_id.split("_")[0])
+
+
+def test_retrieve_caravan_forcing_no_basin_id(
+    tmp_path: Path, mock_retrieve: mock.MagicMock
+):
+    test_files_dir = Path(__file__).parent / "forcing_files"
+    tmp_camels_dir = tmp_path / "camels"
+    copytree(test_files_dir, tmp_camels_dir)
+
+    msg = (
+        "You have to specify a basin ID to be able to generate forcing from" " Caravan."
```

Comment:
```suggestion
        "You have to specify a basin ID to be able to generate forcing from Caravan."
```
unnecessary string concatenation